docs/docs/development/hardware-integration/dongle.mdx

```diff
@@ -0,0 +1,302 @@
+
+</TabItem>
+<TabItem value="base">
+Copy the physical layout node into your `my_keyboard_dongle.overlay` file. Make sure the matrix transform is assigned to it, and select it in the `chosen` node.
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+
+    physical_layout0: physical_layout_0 {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_transform>;
+        keys = <...>; // Long list of key positions
+    };
+};
+```
+
+Make sure that the labels `physical_layout0` and `default_transform` match those of the physical layout node and the matrix transform respectively.
+
+If there are multiple physical layouts, you need only copy the one that applies to your keyboard.
+
+</TabItem>
+<TabItem value="imported">
+Include your desired layout at the top of the file. Then assign your matrix transform to it, taking care to match up the node labels correctly.
+
+```dts
+#include <layouts/<layout_name>.dtsi>
+
+&physical_layout0 {
+    transform = <&default_transform>;
+};
+```
+
+In this case your node label will _not_ be `physical_layout0`, make sure to adjust it accordingly.
+
+Finally, select your physical layout in the chosen node.
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &kscan0;
+        zmk,physical-layout = &physical_layout0;
+    };
+};
+```
+
+</TabItem>
+</Tabs>
+
+</TabItem>
+<TabItem value="nostudio">
+Add a barebones [physical layout](physical-layouts.md) without the `keys` property to your `my_keyboard_dongle.overlay` file.
+
+```dts
+/ {
+    physical_layout0: physical_layout_0 {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_transform>;
+    };
+};
+```
+
+Make sure that the `transform` property uses the same label as your matrix transform. Then select the physical layout in the same `chosen` node that was previously created:
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;  // New item added
+    };
+};
+```
+
+</TabItem>
+</Tabs>
+
+## Building the Firmware
+
+:::warning
+Before flashing your new firmware, you need to flash `settings_reset` [firmware](../../troubleshooting/connection-issues.mdx#acquiring-a-reset-uf2) on all devices to ensure they can pair to each other.
+:::
+
+Add the appropriate lines to your `build.yml` file to build the firmware for your dongle. Also add some CMake arguments using `cmake-args` to the existing parts of your keyboard, turning them into peripherals for your dongle.
+
+```yaml
+include:
+  # -----------------------------------------
+  # Your other keyboard parts here
+  # -----------------------------------------
+  # Change the board appropriately, you can use any board
+  - board: nice_nano_v2
+    shield: my_keyboard_dongle
```

Comment:
What about ZMK Studio?

Should we add cmake-args to make it work?

Something like:
```
  - board: nice_nano_v2
    shield: my_keyboard_dongle
#    cmake-args: -DCONFIG_ZMK_STUDIO=y  # For ZMK Studio to work
```

I may be wrong, because mine is not compiling with that arg and I am trying to figure out why